corehq/apps/domain/models.py

```diff
@@ -1047,6 +1059,46 @@ def update_calculations(cls, domain, property_to_update):
         obj.save(update_fields=[property_to_update])
 
 
+class AllowedUCRExpressionSettings(models.Model):
+    """
+    Model contains DTE(aka UCR) expressions settings for a domain.
+    The expressions defined in RESTRICTED_DTE_EXPRESSIONS are not generally available yet.
+    But these expressions are enabled by default on every domain so that current flow does not change.
+    If any expression's usage is to be restricted on any domain
+    then the  Expressions should be explicitly removed from
+    Domain settings page on HQ.
+    """
+
+    domain = models.CharField(unique=True, max_length=256)
+    allowed_ucr_expressions = ArrayField(
+        models.CharField(max_length=32, choices=RESTRICTED_UCR_EXPRESSIONS),
+        default=get_default_ucr_expressions
+    )
+
+    def save(self, *args, **kwargs):
```

Comment:
Why is this necessary? I would have thought a standard array field would allow us to use the standard save function.

Reply:
This was required to clear cache of `get_allowed_ucr_expressions`

Reply:
The standard way to do this would be to use a `post_save` signal handler

Reply:
Ohh yes!! I will update it.

Reply:
https://github.com/dimagi/commcare-hq/pull/31103/commits/05855b3b38f76c40a25775120abfa1c482dea67e